### PR TITLE
Metastation 2.0

### LINKED ISSUE #7429: Metastation: Bridge's camera console
**Problem Description**:
You cannot view any cameras on the bridge.

**What did you expect to happen**: 
Able to see a list and select a camera to view

**What happened instead**
Shows a map of the station with no available cameras

**Why is this bad/What are the consequences:**
Cannot search or locate the problem.

**Steps to reproduce the problem**:
Be at bridge, open camera console.

### LINKED ISSUE #7430: *Metastation* Missing Telecomms Key
**Problem Description**:
The decryption key note does not appear to spawn in the RD's office, assuming it wasn't stolen before I joined.

**What did you expect to happen**: 
On Box Station the key spawns on the RD's desk at round start.

**What happened instead**:
See description.

**Why is this bad/What are the consequences:**
The RD needs that key to perform a part of their job, namely, monitoring the PDA server when necessary, as well as keeping the key itself safe.

**Steps to reproduce the problem**:
Check the RD's desk at round start.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Anything else you can tell us.

### LINKED ISSUE #7431: *Metastation* Incomplete RD access
**Problem Description**:
The RD does not have access to the research outpost shuttle.

**What did you expect to happen**: 
For the RD to have access to the shuttle.

**What happened instead**:
My access was denied when I tried to board the shuttle.

**Why is this bad/What are the consequences:**
The RD needs access to all parts of science, to make sure the scientists are being productive, and also not blowing anything up.

**Steps to reproduce the problem**:
Attempt to board the research outpost shuttle as the RD.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Anything else you can tell us.

I do not know if regular scientists have access or not, but if not, they should also have access.

### LINKED ISSUE #7432: [Metastation] Chapel shutters are transparent
**Problem Description**:
You can see through the shutters in the chapel when they are closed.

**What did you expect to happen**: 
I expected the shutters to block vision when closed.

**What happened instead**:
I could see through them when shut.

**Why is this bad/What are the consequences:**
Unable to get privacy from any space carp.

**Steps to reproduce the problem**:
Press either of the shutter buttons in the chapel.
Watch as you can see through the other side.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: The round was testing the new atmos system, I didn't see if the closed shutters still blocked atmos.

### LINKED ISSUE #7433: [Metastation] Engineering secure storage blastdoors don't block air
**Problem Description**:
Engineering secure storage blast shutters don't block atmospherics.

**What did you expect to happen**: 
Secure storage blastdoors to be air tight.

**What happened instead**:
Gasses freely pass when blastdoors closed

**Why is this bad/What are the consequences:**
Giant plasma fire in engineering.

**Steps to reproduce the problem**:
Do air stuff on one side with doors closed, watch it pass.

### PR DESCRIPTION
**Pull request by SamHPurp**
With the merge of this PR and related PRs #7396, PR #7388... and following already merged PR #7376, #7158, #7126.... MetaStation will be at the standard where we should consider it for an active **rotation**. In my opinion.

---------

This PR this makes the following **improvements**, **fixes**, and **tweaks**... culminating in MetaStation v2.0, finally at a Paradise-level quality.

* Dorm rooms and communal areas now have glass doors, and unnecessary window tints removed.
* Gloves and masks have been added to robotics.
* Botany is no longer all-access....
* Morgue has been given a makeover (image now defunct due to full rework of medbay)
* Added fire-locks to key areas, Fixes #7433
* Sets access requirements correctly on the path to the Science Outpost Shuttle, Fixes #7431
* Aligns the monitor encryption key with Box Station, moving it from the CE's office to the RD's office. Fixes #7430
* Camera network monitors now function properly. Fixes #7429. (Relatedly #7428 will be fixed by the merging of #7396)
* Church shutters block vision again, Fixes #7432
* Adds the missing armoury shutter button
* Adds dogbeds to the HoS and CMO's offices.
* Readds Sergeant Araneus.
* Adds a processing area to the Brig.

Changes as of 7th of June:
* Revamps much of the brig, while keeping the same overall format
~~Image removed, new one further down~~
* Adds the Gamma Armoury
* Adds two additional brig cells, totaling five
* Adds a morgue slab to the Brig Phys office
* Expands the cryopods, bringing up the number of cryosleepers up to a more reasonable number
* Engine Containment is now protected by Blast Doors, not Shutters.
* Fixes the blast doors in the Execution Chamber

Changes as of 10th of June:
* Alters room layout of the brig to give the Warden a direct path into the armory.
image

Changes as of 20th of June:
Entirely tore out and redid all of Medbay.
[Here is old medbay....](https://cdn.discordapp.com/attachments/145700045008797697/326693767111376896/unknown.png)

And the new one:
image

Basic tenants I tried to adhere to while I built it: Expand reception, add more ways to get into Medical. Expand the walkways. Spread out the jewels so a single bomb doesn't take out the whole place... Add a direct access between Cloner and Morgue (to make @ZomgPonies happy ;) ). Make surgery not a massive pile of shit.

This also now includes the edits previously separated into #7388 for all the reasons stated on that original PR.

---

:tada:

We ran 4 MetaStation tests yesterday totaling about 7 hours of total time. I ran surveys the whole time. It went extremely well, with only minor bug reports.

:cl: Purpose2
add: MetaStation: Adds a brand new Medbay!
add: MetaStation: Adds the dogbeds to the CMO/HoS office
add: MetaStation: Adds Sergeant Araneus back to the HoS's office. HoS beware.
add: MetaStation: Adds a processing area to the brig.
add: MetaStation: Adds the Gamma Armory.
add: MetaStation: Adds two additional brig cells
add: MetaStation: Adds a morgue slab to the Brig Phys office
add: MetaStation: Morgue has a swanky new office, directly connected to Cloning.
add: MetaStation: Robotics gets gloves and masks.
add: MetaStation: Adds shutters to R&D and Genetics.
add: MetaStation: Adds a Paramedic's office.
add: MetaStation: Adds a fully function containment grid.
tweak: MetaStation: Moves Monitor Encryption key to the RD's office again
tweak: MetaStation: Dorms have glass doors and fewer tinted windows
tweak: Metastation: Engine Containment now has Blast Doors instead of Shutters.
fix: Metastation: Execution chamber blast doors are no longer see-through.
fix: Metastation: Only the appropriate plastic flaps should be see-through.
fix: MetaStation: Botany is no longer all-access
fix: MetaStation: Science gets access to the Science outpost again... for now...
fix: MetaStation: Fixes camera network monitors
fix: MetaStation: Church shutters block vision again
fix: MetaStation: Adds the missing Armory shutter button.
del: MetaStation: Removes tinted windows from R&D & Genetics
/ :cl:

### CODE REVIEW

**Comment by Anticept** [comment]
PA room needs blast doors instead of shutters.

**Comment by SamHPurp** [comment]
@Anticept doesnt' really NEED them, they aren't mechanically any different other than you can't door slam people with them.

I'd be happy to change them, if people agreed.

**Comment by FlattestGuitar** [comment]
Look at that spider. That's a happy spider.

Solid PR. I oppose nothing here.

**Comment by scrubmcnoob** [comment]
Doesn't sec only have three cells on this station?

**Comment by Anticept** [comment]
@SamHPurp more bomb resistance, for what its worth.

**Comment by EvadableMoxie** [comment]
I have serious concerns about Metastation's Medbay, and it's able to properly handle Paradise's population.  Where would be the best place to provide that feedback?  It's long, because there's a lot wrong with Meta-Medbay.  Basically everything about it. Except virology, which is fine.

**Comment by Anoonki** [comment]
But will araneus be a dick and smash the HoS' desk every round on this map?

**Comment by Da-Dman234** [comment]
`Doesn't sec only have three cells on this station? This could be an isssue if we wanted to mainstream this map.`

This. Good work with what's been done but there are still a number of core issues with the map. I really dislike the map as a whole, but I'll try to leave that out of this. From what I've seen (and even made a suggestions thread about) the biggest issue is the amount of cells sec has, and the AI upload. Sec needs more cells for obvious reasons, and the current way the AI upload is, as soon as someone walks in if the turrets are on lethal both consoles are immediately destroyed, which again, is a problem for obvious reasons.

**Comment by SamHPurp** [comment]
@Da-Dman234 , @EvadableMoxie and @scrubmcnoob I'm looking into this areas of concern now.

**Comment by EvadableMoxie** [comment]
Well, since there doesn't seem to be a better place for it, here's my Medbay feedback:

The biggest and most glaring issue is that almost all traffic must be funneled through a single 2 tile wide door, which then immediately opens to the Medbay fridge (A high traffic area people loiter in), and then travel down a hall that chokes into 2 tiles again due to a NanoMed Vendor.  That just isn't acceptable for Paradise's population.   

This is further compounded by just about everything except cloning all being in the same direction.  Cryo and Sleepers are right next to each other and surgery actually requires going through and past Cryo.  This further congests things as compared to the current Medbay.  It's also an antag's wet dream as a single syndi bomb could potentially take out every sleeper and cryotube in Medbay plus surgery.  Or a bomb in the entrance would leave people with no way in or out, as compared to the current Medbay which at least has a side door.

Genetics is tucked away with just the door and 1 tile of window near it to view it from Cloning.  This pretty much guarantees genetics is going to ignore the cloner (Not that they ever pay attention to it anyway, I suppose) and makes it much harder for doctors to get humanized monkeys or clean S.E. 

Surgery is another huge issue.  Obviously having a body scanner in a corner that requires you to put the patient in, walk around the table to view the console, and then walk back to get the patient out is completely unacceptable, but that's just the tip of the iceberg.  The top OR has no scanner at all, requiring you to completely leave the OR and move through the staging room. 

Observation rooms for ORs are in general bad when observation is the only purpose they serve, and doubly so when they don't allow direct access in or out of the surgery room itself.   On the Cyberaid, the observation room for the ORs is also surgery access, a waiting room, a recovery ward, and a storage room.  It works because it does all of those things well in limited amounts of space.  Further to that, because the entrance is also the observation room, it means as you enter you get a clear view of both ORs and know at a glance which one is available.  And if neither are, you don't have to go anywhere because you're already in the waiting room.

On Meta-station the ORs are closed in, you can't tell if they are in use until you've already moved to the doors and looked inside, which is far less convenient.  And then when you are waiting the waiting area doesn't have a clear view for you to see the progress of ongoing surgeries. 

And even getting into the ante-room to surgery can be a chore since it's a 1 tile choke that requires going past Cryo.  On a server where surgery isn't done much it made sense to have it tucked away in the back, but on para surgery is done very often and it needs to be something with easy access.

The surgery rooms themselves don't make efficient use of space at all.  I understand this is because originally there was only 1 OR so a second had to be shoe-horned in but we really don't need 7 tiles worth of tables per OR.  The current ORs have 5 and 3 would probably do.

And speaking of that, the entire Medbay itself does not make efficient use of space.  Things like the storage room or the CMO's room seem to be sized and shaped completely arbitrarily.  The CMO's office is huge, yet reception is stuffed into a tiny 2x2 room. And why is the CMO's office immediately next to the entrance, despite being a place only 1 person will ever use? That space is extremely valuable, it should be where something like the sleepers or cryo is.  

The maintenance corridor between the CMO's office and chem is completely useless, it leads to and from nowhere and it's only use it to assist antags in breaking into chem and the CMO's office, which has never been something we've needed on the Cyberaid.  Genetics has an entire front area across from science with a desk which is completely useless and will never be used or manned.  The isolation rooms, one of the least used parts of Medbay, are right smack dab in the middle of it, blocking access to surgery except through Cryo.

So here are my suggestions on how to fix all this:

Get rid of the observation room and expand the ORs a bit.  Put a scanner in each one. Reduce the number of tiles spent on table space.  Change the ante-room to surgery so that it has grates and glass separating it from the ORs, allowing a clear view into the ORs from the waiting room.  Move the isolation rooms elsewhere so access to surgery doesn't require going through cryo.  

Shrink genetics down a bit so the top of the genetics room and the cloning room are level (seriously, what is it with the room design on this station?) then change the barrier between the two to be glass and grates so Genetics can see Cloning and Doctors can make requests more easily.

Move the CMO office to where the sleepers are now and open up that whole area and put the sleepers in there somewhere. Get rid of the useless genetics desk area, get rid of the useless maintenance hallway, and shrink chem a bit to make room for all this.  

Ideally, additional entrances to Medbay could be added.  One to the top where storage is now by shrinking/relocating it, and another down near escape across from science would be nice.  This isn't absolutely needed if the current reception is expanded, though.

I realize this is a lot to do but the fact is Metastation was not designed for our level of population, nor for for Medical system where injures are more persistent and surgery is done more often.  Because of this, it needs an overhaul.

**Comment by SamHPurp** [comment]
High quality feedback @EvadableMoxie . I don't think I disagree with any of it.

I'll have a play.

**Comment by ModDownloading** [comment]
I agree with all of this pull report except for the glass doors on dorms, I think those are supposed to be more private if you want to, say, change your jumpsuit in there and not have random chaps passing by and seeing you in underclothes. Oh, and if you would oblige, I would appreciate moving the Autodrobe and Hatlord/Shoelord/Suitlord (and the ArtVend and Laptop Vendor too, because I couldn't find them) to the locker room.

Other than that, this looks absolutely magnificent and I am greatly looking forward to when this map enters rotation!

**Comment by SamHPurp** [comment]
@ModDownloading the bedrooms are absolutely still not glass doors, its the main walkways.

**Comment by ModDownloading** [comment]
Ah, when you said dorm rooms and communal areas have glass doors, I must have misinterpreted that as the actual dormitory doors, sorry about that.

**Comment by deep-fried-faceplant** [comment]
Remove the giant central desk, and the top middle desk, and remove all those windows on the north and south walls, or add shutters to everything. If metastation HAS to have processing, I would much rather have a processing area with multiple rooms, rather than one large room which will no-doubt turn into a quagmire of lost items, criminals, and bucklecuffed innocents. I was happy to see that metastation didn't have processing, it's only needed for serious crimes or people possibly headed for perma. Having processing encourages officers to dump prisoners off in processing, then leave them for the warden, under the impression that officers are not allowed to brig prisoners. In the past months, this misconception of officers being unable to brig their quarries has driven me insane. But, I digress.

Overall, I'm not happy with this design for processing.

**Comment by SamHPurp** [comment]
@deep-fried-faceplant I'm happy to make some changes and make it into a smaller thing, re-work it a bit... but part of the idea of it was that it was a bit of a clusterfuck ;)

As for Processing vs no Processing... I'd suggest you setup a thread on the forums and we can discuss the merits, as long as the discussion also includes sensible concessions for balance.

**Comment by deep-fried-faceplant** [comment]
Looking at processing gave me a little heart attack, please don't scare me like that. I still have flashbacks to three screaming vox in processing...

**Comment by SamHPurp** [comment]
Revamps much of the brig, while keeping the same overall format
image

Adds the Gamma Armoury
Adds two additional brig cells, totaling five
Adds a morgue slab to the Brig Phys office
Expands the cryopods, bringing up the number of cryosleepers up to a more reasonable number
Engine Containment is now protected by Blast Doors, not Shutters.
Fixes the blast doors in the Execution Chamber

--------

TODO: Revamp medical.

File: code/game/machinery/doors/brigdoors.dm [hunk 1]
```diff
@@ -65,7 +65,7 @@
 	Radio.follow_target = src
 
 	pixel_x = ((dir & 3)? (0) : (dir == 4 ? 32 : -32))
-	pixel_y = ((dir & 3)? (dir ==1 ? 24 : -32) : (0))
+	pixel_y = ((dir & 3)? (dir ==1 ? 32 : -32) : (0))
```
Comment (SamHPurp) [inline code/game/machinery/doors/brigdoors.dm]: I couldn't see why this was done at `24`. It looks awful at `24`, and is inconsistent from all other directions.

This won't effect Box, as no north-facing ones are used.

**Comment by deep-fried-faceplant** [comment]
Because this PR is a brig remap, and because I keep kicking myself in the ass for not contributing to the boxstation brig remap, I really want to give my input on why processing is almost always terrible, and how it can be fixed through map changes.

The purpose of the processing room is to have a room where prisoners can be safely searched, told their crimes, questioned, then thrown into a cell. Usually, the processing process can be done in a cell, but there's nothing to stop officers from using the processing room anyways. What happens in that room is what makes security so damn stressful.

Usually, processing is one room with several tables, implying that multiple people can be effectively processed in the same room. However, because it's hard to zone out other people talking in chat, especially when there's a greytider three tiles away screaming at the top of his lungs that he's innocent, processing quickly becomes a chatroom going far too fast. This makes processing prisoners much more difficult. It's why I don't like windows in processing at all, because processing should have no outside interference. Being able to hear someone's conversation in the hallway, while you're trying to listen to a prisoner, while the warden is issuing a sentence, is very, very annoying.

Multiple people tend to be in processing. The arresting officer, the prisoner, the warden, sometimes the HoS, an IAA, the captain, ect. It can get very crowded very fast. More room helps.

People leave shit in processing all the time. Oh lordy lordy, people have been accidentally executed because someone didn't put a traitor's gear bag into evidence, and a well meaning officer thought that the bag was some innocent guy's bag. I've seen this multiple times. Random desks and chairs make processing a clusterfuck. While normally, the station is supposed to be a metal deathtrap in the sky, processing should be exempt from the rule. Processing is where hopes and dreams go to die, where players get banned for good intentions and prisoners go SSD because they've been in there for 30 minutes. Making tables that are meant for security to sit at only have stools would make officers NOT cuff people to those tables.

Attached here is an example of what I think is a good processing room. It's two rooms, each with one table for the prisoner, and one long table for the evidence. There are no windows to the outside, there's wiggle room for people to stand against the wall and watch, and there are chairs in a quasi-lobby for people to wait their turn. I'm not suggesting we implement this crude draft of processing to metastation, it's only here for demonstrative purposes.
super crude processing draft

**Comment by SamHPurp** [comment]
You are not wrong. It is also an intentional design decision. By this point the antagonist is largely boned, and the chaos of a bull pen aids them. ;)

**Comment by deep-fried-faceplant** [comment]
Then remove the airlocks to each room, so if a traitor gets uncuffed, they can still get out without having to bust through two airlocks. Intentional bad design is a horrible way to buff antagonists. Processing is not the place for an antagonist to make their last stand unless they have a freedom implant or outside help. A captured antagonist in processing should use their words, rather than brute force, if the antagonist's duel with security has culminated to a bucklecuffing in the brig.

Processing has a very real mechanical purpose, almost OOC in its importance and regulation. I want processing to be good for what it was meant to be: processing. Currently, one of the key problems with security is processing glut, when prisoners get stuck in processing for more then ten minutes, and items are left strewn across the room. This isn't an intentional feature, it's a problem bred from horrible design and slack security. Processing​ glut can get people banned. Processing glut burns out security players. Nobody enjoys processing glut except for the rare traitor caught up in it who had miraculously managed to escape their cuffs. The state of processing reflects the state of security, which in turn reflects upon the server. If processing goes to shit, security is called shitcurity, and it builds a reputation for Paradise that we have shit security. Please, don't make processing anymore difficult than it really has to be. I'm pleading with you here.

**Comment by SamHPurp** [comment]
Altered the layout following feedback from @FreeStylaLT - a direct path to the Armory from the Warden's office 
image

**Comment by uraniummeltdown** [comment]
That Medbay's looking pretty good. I do miss the maint entrance into CMO, MedChem, and Medbay but it's probably for the best

As for the station name I think something from https://en.wikipedia.org/wiki/Vermes_in_the_10th_edition_of_Systema_Naturae would be good.

NSS Nautilus, Helix, Hydra, Asterias; all sound like great names

**Comment by SamHPurp** [comment]
If you haven't yet - please have a quick look at http://www.strawpoll.me/13239561 and vote! 👍

**Comment by SamHPurp** [comment]
A week on, the poll's results have been given to the Headmins to make a final decision.

**Comment by SamHPurp** [comment]
@tigercat2000 @Fox-McCloud @Crazylemon64 conflicts gone lets dooooo thissssss https://www.youtube.com/watch?v=SFkdcQgNJHo

**Comment by SamHPurp** [comment]
Also to note: I haven't changed the name on this PR... I was given an estimate of "three working years" on getting back to me on the name.... so this'll have to come later.

**Comment by tigercat2000** [comment]
Final map image:

**Pull request merged**